### PR TITLE
Voting for comments and  posts with tests

### PR DESCRIPTION
**Pull request by Timgur**

### CODE REVIEW

**Comment by Timgur** [comment]
@ecto @Polyrhythm

**Comment by pastudan** [comment]
@Timgur just curious, why do we version control the build/ directory?

**Comment by Timgur** [comment]
@pastudan not sure why .... please see this https://github.com/Imgur/imgur.js/pull/18

**Comment by pastudan** [comment]
@Timgur looked at that RIGHT after I left my comment. My bad dude :p

File: build/imgur.js [hunk 1]
```diff
@@ -96,18 +96,61 @@
...
+        REASON_DOES_NOT_BELONG_ON_IMGUR: 1,
         get: function get(hash) {
             var path = '' + this.path + '/' + hash;
             var options = utils.buildOptions(this.apiUrl, path, 'get');
 
             return this.imgurAPICall(options);
         },
-        report: function report(hash, reason) {
+        report: function report(hash) {
+            var reason = arguments[1] === undefined ? this.REASON_DOES_NOT_BELONG_ON_IMGUR : arguments[1];
```
Comment (pastudan) [inline build/imgur.js]: good place for my favorite js syntactic sugar:
```javascript
var reason = arguments[1] || this.REASON_DOES_NOT_BELONG_ON_IMGUR
```

Comment (Timgur) [inline build/imgur.js]: this is the compile build......

**Comment by pastudan** [comment]
:+1:

File: build/imgur.js (hunk 1 above)
Comment (pastudan) [inline build/imgur.js]: Doh. So that means es6 allows default parameters. Badass!

**Pull request merged**